### PR TITLE
Improving error message when app doesn't exist

### LINKED ISSUE #3052: fatal errors when trying to read some metadata from an unknown app
## Version of Singularity:

3.1.0

<pre>
[justin@ip-172-31-94-96 inspect]$ singularity inspect --app bob -H alpine_latest.sif
FATAL:   exit status 127
[justin@ip-172-31-94-96 inspect]$ singularity inspect --app bob -r alpine_latest.sif
FATAL:   exit status 127
[justin@ip-172-31-94-96 inspect]$ singularity inspect --app bob -t alpine_latest.sif
FATAL:   exit status 127
</pre>

I would not expect to see FATAL, but more like 'App 'bob' not found'. Need to improve the message anyways.

### PR DESCRIPTION
**Pull request by vsoch**
Signed-off-by: Vanessa Sochat <vsochat@stanford.edu>

**Description of the Pull Request (PR):**

This pull request will adjust the inspect command, ensuring that it exits with a clear message if the app the user asked for does not exist. Here is a quick example - we pull a container that doesn't have any apps, and ask for one that doesn't exist:

```bash
$ singularity pull docker://busybox
$ singularity inspect --app bob busybox_latest.sif 
FATAL:   exit status 2: App bob does not exist.
```
It's still fatal because we exit out, but minimally the user knows what is going on.

The container works with inspect as expected:

```bash
$ singularity inspect busybox_latest.sif 

{
	"org.label-schema.build-date": "Friday_22_March_2019_16:52:46_EDT",
	"org.label-schema.schema-version": "1.0",
	"org.label-schema.usage.singularity.deffile.bootstrap": "docker",
	"org.label-schema.usage.singularity.deffile.from": "busybox",
	"org.label-schema.usage.singularity.version": "3.1.0-rc2.505.g3dd8344.dirty"
}
```
Here is a recipe that has a simple app:

```
Bootstrap: docker
From: busybox:latest

%apprun jujuboo
    echo "JOOOO!"
```

Building, and inspecting an app that doesn't exist:

```bash
$ sudo singularity build joo.sif Singularity
$ singularity inspect --app doesntexist joo.sif 
FATAL:   exit status 2: App doesntexist does not exist
```

I chose an [error code](https://mariadb.com/kb/en/library/operating-system-error-codes/) of 2 to indicate it doesn't exist.

Here is does exist:

```bash
$ singularity inspect --app jujuboo --labels joo.sif

{
	"SCIF_APP_NAME": "jujuboo"
}
```
There is still a bit of weirdness going on with what is given for default labels, that can be addressed in another PR.

**This fixes or addresses the following GitHub issues:**

- Fixes #3052 

Attn: @singularity-maintainers

### CODE REVIEW

**Review by mem — APPROVED** [review]
Thanks!

**Comment by vsoch** [comment]
hey @mem I'm not familiar with the command and testing setup - how do we fix this?

```
make: Entering directory `/home/circleci/go/src/github.com/sylabs/singularity/builddir'
 TEST sudo go test [unit]
?   	github.com/sylabs/singularity/cmd/bash_completion	[no test files]
?   	github.com/sylabs/singularity/cmd/docs	[no test files]
# github.com/sylabs/singularity/cmd/internal/cli
cmd/internal/cli/inspect_linux.go:279: Fatalf format %s has arg cmd of wrong type *os/exec.Cmd
FAIL	github.com/sylabs/singularity/cmd/internal/cli [build failed]
```

File: cmd/internal/cli/inspect_linux.go [hunk 1]
```diff
@@ -266,12 +276,12 @@ func getFileContent(abspath, name string, args []string) (string, error) {
 
 	cmd, err := exec.PipeCommand(starter, []string{procname}, Env, configData)
 	if err != nil {
-		sylog.Fatalf("%s", err)
+		sylog.Fatalf("%s: %s", err, cmd)
```
Comment (jscook2345) [inline cmd/internal/cli/inspect_linux.go]: I think this needs to be cmd.Args, not cmd itself.

Comment (vsoch) [reply cmd/internal/cli/inspect_linux.go]: Thanks @jscook2345 I'll give this a go!

**Comment by vsoch** [comment]
Woo! Travis finished and we're green!!

**Comment by GodloveD** [comment]
Thanks @vsoch !

**Pull request merged**